Assets/Scripts/Room/Chat/ChatCommandHandler.cs

```diff
@@ -0,0 +1,777 @@
+            }
+        }
+    }
+
+    /// <summary>
+    /// Revive player with Id
+    /// </summary>
+    /// <param name="playerIdString"></param>
+    private static void RevievePlayer(string playerIdString)
+    {
+        if (isMasterClient)
+        {
+            int playerId;
+            if (int.TryParse(playerIdString, out playerId))
+            {
+                var player = playerList.FirstOrDefault(p => p.ID == playerId);
+
+                if (playerList.Any(p => p.ID == playerId))
+                {
+                    var message = FormatSystemMessage($"Player {playerId} has been revived.");
+                    instance.chatRoom.AddMessage(message);
+                    instance.photonView.RPC("RespawnRpc", player);
+                }
+            }
+            else
+            {
+                instance.chatRoom.OutputErrorPlayerNotFound(playerIdString);
+            }
+        }
+        else
+        {
+            instance.chatRoom.OutputErrorNotMasterClient();
+        }
+    }
+
+    /// <summary>
+    /// Spectate player with Id
+    /// </summary>
+    /// <param name="playerIdString"></param>
+    private static void SpectatePlayer(string playerIdString)
+    {
+        int playerId;
+        if (int.TryParse(playerIdString, out playerId))
+        {
+            foreach (GameObject gameObject in GameObject.FindGameObjectsWithTag("Player"))
+            {
+                if (gameObject.GetPhotonView().owner.ID == playerId)
+                {
+                    Camera.main.GetComponent<IN_GAME_MAIN_CAMERA>().setMainObject(gameObject, true, false);
+                    Camera.main.GetComponent<IN_GAME_MAIN_CAMERA>().setSpectorMode(false);
+                }
+            }
+        }
+        else
+        {
+            instance.chatRoom.OutputErrorPlayerNotFound(playerIdString);
+        }
+            
+    }
+
+    /// <summary>
+    /// Output number of titan Collisions
+    /// </summary>
+    private static void OutputCollisions()
+    {
+        int collisions = 0;
+        foreach (MindlessTitan titan in instance.getTitans())
+        {
+            if (titan.IsColliding)
+            {
+                collisions++;
+            }
+        }
+        var message = collisions.ToString();
+        instance.chatRoom.AddMessage(message);
+    }
+
+    /// <summary>
+    /// Set field of view
+    /// </summary>
+    /// <param name="input"></param>
+    private static void SetFov(string input)
+    {
+        int fov;
+        if (int.TryParse(input, out fov))
+        {
+            Camera.main.fieldOfView = fov;
+            var message = $"Field of vision set to {fov}.";
+            instance.chatRoom.AddMessage(message);
+        }
+        else
+        {
+            instance.chatRoom.OutputErrorMessage("Fov has to be a number");
+        }
+    }
+
+    /// <summary>
+    /// Enter/Exit spectate mode
+    /// </summary>
+    private static void EnterExitSpecMode()
+    {
+        string message;
```

Comment:
Rename to `ToggleSpecMode()`.
```cs
// This will be more readable when we improve the settings.
settings[0xf5] = ((int)settings[0xf5] == 1 ? 0 : 1);
bool specMode = settings[0xf5] == 1;
instance.EnterSpecMode(specMode);
string message = FormatSystemMessage(specMode ? "You have entered spectator mode." : "You have exited spectator mode.");
instance.chatRoom.AddMessage(message);
```